/////////
修改conv7.cpp文件
AlgoIterator<cudnnConvolutionFwdAlgoPerf_t>(args, benchmark).try_all(

  [&](const cudnnConvolutionFwdAlgoPerf_t &fwdAlgPerf){

   Tensor workspace = allocate_workspace(fwdAlgPerf.memory, input);



   // update convDesc mathType since cudnn 7.4+ now requires both algo + mathType to figure out

   // whether to use Tensor core kernels or not

   // See Note [behavior of cudnnFind and cudnnGet]

   ASSERT_CORRECT_PRECISION(fwdAlgPerf.mathType);

   AT_CUDNN_CHECK_WITH_SHAPES(cudnnSetConvolutionMathType(args.cdesc.mut_desc(), fwdAlgPerf.mathType), args);



   Constant one(dataType, 1);

   Constant zero(dataType, 0);

   std::cout<<static_cast<int>(fwdAlgPerf.algo)<<std::endl;

   AT_CUDNN_CHECK_WITH_SHAPES(cudnnConvolutionForward(

​     args.handle,

​     &one, args.idesc.desc(), input.data_ptr(),

​     args.wdesc.desc(), weight.data_ptr(),

​     args.cdesc.desc(), CUDNN_CONVOLUTION_FWD_ALGO_WINOGRAD, workspace.data_ptr(), fwdAlgPerf.memory,

​     &zero, args.odesc.desc(), output.data_ptr()),

​    args, "Forward algorithm: ", static_cast<int>(CUDNN_CONVOLUTION_FWD_ALGO_WINOGRAD), "\n");

​    std::cout<<static_cast<int>(fwdAlgPerf.algo)<<std::endl;

   }

 );
/////////////////////

///////////////////////
运行结果


test_conv1
1
0
2
terminate called after throwing an instance of 'c10::Error'
  what():  Unable to find a valid cuDNN algorithm to run convolution
Exception raised from try_all at /public/home/hpc2316085405032/zip/pytorch/aten/src/ATen/native/cudnn/Conv_v7.cpp:532 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7f50d9bba2e7 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, char const*) + 0x68 (0x7f50d9b70ba0 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #2: <unknown function> + 0xed1993 (0x7f50bd665993 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #3: <unknown function> + 0xec7b9d (0x7f50bd65bb9d in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0xec8050 (0x7f50bd65c050 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0xec524b (0x7f50bd65924b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #6: at::native::cudnn_convolution(at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long, bool, bool, bool) + 0x96 (0x7f50bd6597f6 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #7: <unknown function> + 0x2c2b08e (0x7f50bf3bf08e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #8: <unknown function> + 0x2c47f51 (0x7f50bf3dbf51 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #9: at::_ops::cudnn_convolution::call(at::Tensor const&, at::Tensor const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool) + 0x2be (0x7f50c79ba36e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #10: at::native::_convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long, bool, bool, bool, bool) + 0x1207 (0x7f50c6d55f87 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x25901bd (0x7f50c7d791bd in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x25964fc (0x7f50c7d7f4fc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #13: at::_ops::_convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool, bool) + 0x347 (0x7f50c74e6b67 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #14: at::native::convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long) + 0x3b5 (0x7f50c6d49965 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x258f2bc (0x7f50c7d782bc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #16: <unknown function> + 0x2596368 (0x7f50c7d7f368 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #17: at::_ops::convolution::redispatch(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x17b (0x7f50c74a651b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #18: <unknown function> + 0x3b1c36c (0x7f50c930536c in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #19: <unknown function> + 0x3b1d299 (0x7f50c9306299 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #20: at::_ops::convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x2d7 (0x7f50c74e5937 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #21: <unknown function> + 0x12e3b90 (0x7f50c6accb90 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #22: at::native::conv2d_symint(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x1ff (0x7f50c6d4d36f in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #23: <unknown function> + 0x2786653 (0x7f50c7f6f653 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #24: <unknown function> + 0x27868ed (0x7f50c7f6f8ed in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #25: at::_ops::conv2d::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x271 (0x7f50c7ae71a1 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #26: ./main() [0x412d6c]
frame #27: ./main() [0x409a51]
frame #28: ./main() [0x409f9d]
frame #29: ./main() [0x404fbd]
frame #30: __libc_start_main + 0xf5 (0x7f50bb9d63d5 in /lib64/libc.so.6)
frame #31: ./main() [0x40542f]

run.sh: 行 16: 80428 已放弃               (吐核)./main data ${i}
test_conv2
1
0
2
terminate called after throwing an instance of 'c10::Error'
  what():  Unable to find a valid cuDNN algorithm to run convolution
Exception raised from try_all at /public/home/hpc2316085405032/zip/pytorch/aten/src/ATen/native/cudnn/Conv_v7.cpp:532 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7f30268fb2e7 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, char const*) + 0x68 (0x7f30268b1ba0 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #2: <unknown function> + 0xed1993 (0x7f300a3a6993 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #3: <unknown function> + 0xec7b9d (0x7f300a39cb9d in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0xec8050 (0x7f300a39d050 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0xec524b (0x7f300a39a24b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #6: at::native::cudnn_convolution(at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long, bool, bool, bool) + 0x96 (0x7f300a39a7f6 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #7: <unknown function> + 0x2c2b08e (0x7f300c10008e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #8: <unknown function> + 0x2c47f51 (0x7f300c11cf51 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #9: at::_ops::cudnn_convolution::call(at::Tensor const&, at::Tensor const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool) + 0x2be (0x7f30146fb36e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #10: at::native::_convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long, bool, bool, bool, bool) + 0x1207 (0x7f3013a96f87 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x25901bd (0x7f3014aba1bd in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x25964fc (0x7f3014ac04fc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #13: at::_ops::_convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool, bool) + 0x347 (0x7f3014227b67 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #14: at::native::convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long) + 0x3b5 (0x7f3013a8a965 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x258f2bc (0x7f3014ab92bc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #16: <unknown function> + 0x2596368 (0x7f3014ac0368 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #17: at::_ops::convolution::redispatch(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x17b (0x7f30141e751b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #18: <unknown function> + 0x3b1c36c (0x7f301604636c in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #19: <unknown function> + 0x3b1d299 (0x7f3016047299 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #20: at::_ops::convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x2d7 (0x7f3014226937 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #21: <unknown function> + 0x12e3b90 (0x7f301380db90 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #22: at::native::conv2d_symint(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x1ff (0x7f3013a8e36f in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #23: <unknown function> + 0x2786653 (0x7f3014cb0653 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #24: <unknown function> + 0x27868ed (0x7f3014cb08ed in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #25: at::_ops::conv2d::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x271 (0x7f30148281a1 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #26: ./main() [0x412d6c]
frame #27: ./main() [0x409a51]
frame #28: ./main() [0x40a17d]
frame #29: ./main() [0x404fea]
frame #30: __libc_start_main + 0xf5 (0x7f30087173d5 in /lib64/libc.so.6)
frame #31: ./main() [0x40542f]

run.sh: 行 16: 80703 已放弃               (吐核)./main data ${i}
test_conv3
1
0
2
5
terminate called after throwing an instance of 'c10::Error'
  what():  Unable to find a valid cuDNN algorithm to run convolution
Exception raised from try_all at /public/home/hpc2316085405032/zip/pytorch/aten/src/ATen/native/cudnn/Conv_v7.cpp:532 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7f0af0c702e7 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, char const*) + 0x68 (0x7f0af0c26ba0 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #2: <unknown function> + 0xed1993 (0x7f0ad471b993 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #3: <unknown function> + 0xec7b9d (0x7f0ad4711b9d in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0xec8050 (0x7f0ad4712050 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0xec524b (0x7f0ad470f24b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #6: at::native::cudnn_convolution(at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long, bool, bool, bool) + 0x96 (0x7f0ad470f7f6 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #7: <unknown function> + 0x2c2b08e (0x7f0ad647508e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #8: <unknown function> + 0x2c47f51 (0x7f0ad6491f51 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #9: at::_ops::cudnn_convolution::call(at::Tensor const&, at::Tensor const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool) + 0x2be (0x7f0adea7036e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #10: at::native::_convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long, bool, bool, bool, bool) + 0x1207 (0x7f0adde0bf87 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x25901bd (0x7f0adee2f1bd in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x25964fc (0x7f0adee354fc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #13: at::_ops::_convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool, bool) + 0x347 (0x7f0ade59cb67 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #14: at::native::convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long) + 0x3b5 (0x7f0adddff965 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x258f2bc (0x7f0adee2e2bc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #16: <unknown function> + 0x2596368 (0x7f0adee35368 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #17: at::_ops::convolution::redispatch(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x17b (0x7f0ade55c51b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #18: <unknown function> + 0x3b1c36c (0x7f0ae03bb36c in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #19: <unknown function> + 0x3b1d299 (0x7f0ae03bc299 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #20: at::_ops::convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x2d7 (0x7f0ade59b937 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #21: <unknown function> + 0x12e3b90 (0x7f0addb82b90 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #22: at::native::conv2d_symint(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x1ff (0x7f0adde0336f in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #23: <unknown function> + 0x2786653 (0x7f0adf025653 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #24: <unknown function> + 0x27868ed (0x7f0adf0258ed in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #25: at::_ops::conv2d::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x271 (0x7f0adeb9d1a1 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #26: ./main() [0x412d6c]
frame #27: ./main() [0x409a51]
frame #28: ./main() [0x40a35d]
frame #29: ./main() [0x404e55]
frame #30: __libc_start_main + 0xf5 (0x7f0ad2a8c3d5 in /lib64/libc.so.6)
frame #31: ./main() [0x40542f]

run.sh: 行 16: 80923 已放弃               (吐核)./main data ${i}
test_conv4
1
0
2
5
terminate called after throwing an instance of 'c10::Error'
  what():  Unable to find a valid cuDNN algorithm to run convolution
Exception raised from try_all at /public/home/hpc2316085405032/zip/pytorch/aten/src/ATen/native/cudnn/Conv_v7.cpp:532 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7fcfd8f3d2e7 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, char const*) + 0x68 (0x7fcfd8ef3ba0 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #2: <unknown function> + 0xed1993 (0x7fcfbc9e8993 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #3: <unknown function> + 0xec7b9d (0x7fcfbc9deb9d in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0xec8050 (0x7fcfbc9df050 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0xec524b (0x7fcfbc9dc24b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #6: at::native::cudnn_convolution(at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long, bool, bool, bool) + 0x96 (0x7fcfbc9dc7f6 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #7: <unknown function> + 0x2c2b08e (0x7fcfbe74208e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #8: <unknown function> + 0x2c47f51 (0x7fcfbe75ef51 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #9: at::_ops::cudnn_convolution::call(at::Tensor const&, at::Tensor const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool) + 0x2be (0x7fcfc6d3d36e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #10: at::native::_convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long, bool, bool, bool, bool) + 0x1207 (0x7fcfc60d8f87 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x25901bd (0x7fcfc70fc1bd in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x25964fc (0x7fcfc71024fc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #13: at::_ops::_convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool, bool) + 0x347 (0x7fcfc6869b67 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #14: at::native::convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long) + 0x3b5 (0x7fcfc60cc965 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x258f2bc (0x7fcfc70fb2bc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #16: <unknown function> + 0x2596368 (0x7fcfc7102368 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #17: at::_ops::convolution::redispatch(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x17b (0x7fcfc682951b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #18: <unknown function> + 0x3b1c36c (0x7fcfc868836c in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #19: <unknown function> + 0x3b1d299 (0x7fcfc8689299 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #20: at::_ops::convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x2d7 (0x7fcfc6868937 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #21: <unknown function> + 0x12e3b90 (0x7fcfc5e4fb90 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #22: at::native::conv2d_symint(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x1ff (0x7fcfc60d036f in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #23: <unknown function> + 0x2786653 (0x7fcfc72f2653 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #24: <unknown function> + 0x27868ed (0x7fcfc72f28ed in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #25: at::_ops::conv2d::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x271 (0x7fcfc6e6a1a1 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #26: ./main() [0x412d6c]
frame #27: ./main() [0x409a51]
frame #28: ./main() [0x40a53d]
frame #29: ./main() [0x404e82]
frame #30: __libc_start_main + 0xf5 (0x7fcfbad593d5 in /lib64/libc.so.6)
frame #31: ./main() [0x40542f]

run.sh: 行 16: 81036 已放弃               (吐核)./main data ${i}
test_conv5
1
7
5
0
2
4
terminate called after throwing an instance of 'c10::Error'
  what():  Unable to find a valid cuDNN algorithm to run convolution
Exception raised from try_all at /public/home/hpc2316085405032/zip/pytorch/aten/src/ATen/native/cudnn/Conv_v7.cpp:532 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7f29b442f2e7 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, char const*) + 0x68 (0x7f29b43e5ba0 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libc10.so)
frame #2: <unknown function> + 0xed1993 (0x7f2997eda993 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #3: <unknown function> + 0xec7b9d (0x7f2997ed0b9d in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0xec8050 (0x7f2997ed1050 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0xec524b (0x7f2997ece24b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #6: at::native::cudnn_convolution(at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long, bool, bool, bool) + 0x96 (0x7f2997ece7f6 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #7: <unknown function> + 0x2c2b08e (0x7f2999c3408e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #8: <unknown function> + 0x2c47f51 (0x7f2999c50f51 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cuda.so)
frame #9: at::_ops::cudnn_convolution::call(at::Tensor const&, at::Tensor const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool) + 0x2be (0x7f29a222f36e in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #10: at::native::_convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long, bool, bool, bool, bool) + 0x1207 (0x7f29a15caf87 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x25901bd (0x7f29a25ee1bd in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x25964fc (0x7f29a25f44fc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #13: at::_ops::_convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt, bool, bool, bool, bool) + 0x347 (0x7f29a1d5bb67 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #14: at::native::convolution(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long) + 0x3b5 (0x7f29a15be965 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x258f2bc (0x7f29a25ed2bc in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #16: <unknown function> + 0x2596368 (0x7f29a25f4368 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #17: at::_ops::convolution::redispatch(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x17b (0x7f29a1d1b51b in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #18: <unknown function> + 0x3b1c36c (0x7f29a3b7a36c in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #19: <unknown function> + 0x3b1d299 (0x7f29a3b7b299 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #20: at::_ops::convolution::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, bool, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x2d7 (0x7f29a1d5a937 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #21: <unknown function> + 0x12e3b90 (0x7f29a1341b90 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #22: at::native::conv2d_symint(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x1ff (0x7f29a15c236f in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #23: <unknown function> + 0x2786653 (0x7f29a27e4653 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #24: <unknown function> + 0x27868ed (0x7f29a27e48ed in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #25: at::_ops::conv2d::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::ArrayRef<c10::SymInt>, c10::SymInt) + 0x271 (0x7f29a235c1a1 in /public/home/hpc2316085405032/zip/pytorch/libtorch-cudnn-d3/lib/libtorch_cpu.so)
frame #26: ./main() [0x412d6c]
frame #27: ./main() [0x409a51]
frame #28: ./main() [0x40a71d]
frame #29: ./main() [0x404eaf]
frame #30: __libc_start_main + 0xf5 (0x7f299624b3d5 in /lib64/libc.so.6)
frame #31: ./main() [0x40542f]

run.sh: 行 16: 81748 已放弃               (吐核)./main data ${i}
test_conv6
1
7
7
0.281416
test_conv7
1
1
0.227871
test_conv8
1
1
2.67878
test_conv9
1
1
0.258001
test_conv10
1
1
0.290941
test_conv11
1
7
7
0.218282
test_conv12
1
0
2
5
5
0.159764

